src/Aspire.Hosting/Dashboard/proto/resource_service.proto

```diff
@@ -237,6 +237,9 @@ message Resource {
 
     // The list of relationships for this resource.
     repeated ResourceRelationship relationships = 20;
+
+    // Whether the resource should be visually hidden in the dashboard.
+    bool hidden = 21;
```

Comment:
Is optional bool a thing @jamesnk? We need to distinguish false and not set.

if you have state “Hidden” and Hidden false what does it mean @adamint ?

Reply:
I don't think we want a tri state bool.

IMO a resource is hidden if hidden property is true or the state is hidden (for backwards compatibility).